### PR TITLE
(require|disallow)AnonymousFunctionExpression: account for shorthand …

### LINKED ISSUE #1470: Fix issues for rules about function spacing (ES6)
https://github.com/jscs-dev/babel-jscs/issues/4 was about shorthand methods erroring but it probably affects more than just `requireSpacesInAnonymousFunctionExpression`.

We probably need to check getters, setters, class methods, and object shorthand for all the function spacing rules (similar to the checks in `requireArrowFunctions`).

Similar to #1066 

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/22475358-fix-issues-for-rules-about-function-spacing-es6?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by hzoo**
…methods

Fixes #1470 

I think I still need to do this for the other types and other relevant rules. (Can just add to this PR)

### CODE REVIEW

File: lib/rules/require-spaces-in-anonymous-function-expression.js [hunk 1]
```diff
@@ -89,25 +90,33 @@ module.exports.prototype = {
...
-                        message: 'Missing space before opening curly brace'
-                    });
-                }
+            // if shorthand method
+            if (parent.method) {
+                functionNode = parent.key;
+            }
+
+            // anonymous function expressions only
+            if (node.id) {
```
Comment (hzoo) [inline lib/rules/require-spaces-in-anonymous-function-expression.js]: just going to early return like the disallow rule

Comment (zxqfox) [inline lib/rules/require-spaces-in-anonymous-function-expression.js]: :+1:

**Comment by hzoo** [comment]
I would want to land this and #1494 (esnext fixes) for the next release too @mrjoelkemp

**Comment by mrjoelkemp** [comment]
This is missing an autofix test.

**Comment by hzoo** [comment]
Haven't even had to do those yet haha - done.

**Comment by mrjoelkemp** [comment]
Looks like a test case is missing for this line: 
https://coveralls.io/builds/2972302/source?filename=lib%2Frules%2Frequire-spaces-in-anonymous-function-expression.js#L100.

Once that's resolved, it should be good to merge.

**Pull request closed without merging**